pandas/core/nanops.py

```diff
@@ -1218,7 +1278,12 @@ def _zero_out_fperr(arg):
 
 
 @disallow("M8", "m8")
-def nancorr(a, b, method="pearson", min_periods=None):
+def nancorr(
+    a: np.ndarray,
+    b: np.ndarray,
+    method: str = "pearson",
```

Comment:
this needs to be Union[str, Callable[...]]

Reply:
mypy raises errors

Reply:
then something else is wrong